asserts/asserts_test.go

```diff
@@ -973,3 +973,112 @@ func (as *assertsSuite) TestSequenceForming(c *C) {
 
 	c.Check(asserts.SnapDeclarationType.SequenceForming(), Equals, false)
 }
+
+func (as *assertsSuite) TestHeadersFromSequenceKey(c *C) {
+	headers, err := asserts.HeadersFromSequenceKey(asserts.TestOnlySeqType, []string{"one"})
+	c.Assert(err, IsNil)
+	c.Check(headers, DeepEquals, map[string]string{"n": "one"})
+
+	_, err = asserts.HeadersFromSequenceKey(asserts.TestOnlySeqType, []string{})
+	c.Check(err, ErrorMatches, `sequence key has wrong length for "test-only-seq" assertion`)
+
+	_, err = asserts.HeadersFromSequenceKey(asserts.TestOnlySeqType, []string{""})
+	c.Check(err, ErrorMatches, `sequence key "n" header cannot be empty`)
+}
+
+func (as *assertsSuite) TestAtSequenceString(c *C) {
+	atSeq := asserts.AtSequence{
+		Type:        asserts.ValidationSetType,
+		SequenceKey: []string{"16", "canonical", "foo"},
+		Sequence:    8,
+		Revision:    2,
+	}
+	c.Check(atSeq.String(), Equals, "validation-set canonical/foo/8 at revision 2")
+
+	// Sequence number not set
+	atSeq = asserts.AtSequence{
+		Type:        asserts.ValidationSetType,
+		SequenceKey: []string{"16", "canonical", "foo"},
+		Revision:    asserts.RevisionNotKnown,
+	}
+	c.Check(atSeq.String(), Equals, "validation-set canonical/foo")
+
+	atSeq = asserts.AtSequence{
+		Type:        asserts.ValidationSetType,
+		SequenceKey: []string{"16", "canonical", "foo"},
+		Sequence:    8,
+		Pinned:      true,
+		Revision:    2,
+	}
+	c.Check(atSeq.String(), Equals, "validation-set canonical/foo=8 at revision 2")
+
+	atSeq = asserts.AtSequence{
+		Type:        asserts.ValidationSetType,
+		SequenceKey: []string{"16", "canonical"},
+		Revision:    2,
+	}
+	c.Check(atSeq.String(), Equals, "validation-set ??? at revision 2")
+}
+
+func (as *assertsSuite) TestAtSequenceUnique(c *C) {
+	atSeq := asserts.AtSequence{
+		Type:        asserts.ValidationSetType,
+		SequenceKey: []string{"16", "canonical", "foo"},
+		Sequence:    8,
+		Revision:    2,
+	}
+	c.Check(atSeq.Unique(), Equals, "validation-set/16/canonical/foo")
```

Comment:
Would it make sense to add one more test with a shorter AtSequence? An input like the above ` 		SequenceKey: []string{"16", "canonical"},`

Reply:
Added (although Unique() doesn't validate/care, in the contexts where this is used in real code a valid sequence key is assumed).